File: examples/order/node/src/health/health.controller.ts
import { Controller, Get } from "@nestjs/common";
import { ApiTags, ApiOperation, ApiResponse } from "@nestjs/swagger";

@ApiTags("Health")
@Controller()
export class HealthController {
  @Get("healthz")
  @ApiOperation({ summary: "Liveness probe" })
  @ApiResponse({ status: 200, description: "Service is alive" })
  healthCheck() {
    return { status: "ok" };
  }

  @Get("readyz")
  @ApiOperation({ summary: "Readiness probe" })
  @ApiResponse({ status: 200, description: "Service is ready" })
  @ApiResponse({ status: 503, description: "Service is not ready" })
  readinessCheck() {
    // Add any readiness checks here (database connections, etc.)
    return { status: "ok" };
  }
}

File: examples/order/dotnet/Health/HealthController.cs
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Diagnostics.HealthChecks;

namespace OrdersService.Health;

[ApiController]
[Route("")]
public class HealthController : ControllerBase
{
    private readonly HealthCheckService _healthCheckService;

    public HealthController(HealthCheckService healthCheckService)
    {
        _healthCheckService = healthCheckService;
    }

    /// <summary>
    /// Liveness probe - indicates if the application is running
    /// </summary>
    [HttpGet("healthz")]
    public async Task<IActionResult> Liveness()
    {
        var result = await _healthCheckService.CheckHealthAsync();
        
        if (result.Status == HealthStatus.Healthy)
        {
            return Ok(new { status = "healthy" });
        }

        return StatusCode(503, new { status = "unhealthy" });
    }

    /// <summary>
    /// Readiness probe - indicates if the application is ready to serve requests
    /// </summary>
    [HttpGet("readyz")]
    public async Task<IActionResult> Readiness()
    {
        var result = await _healthCheckService.CheckHealthAsync();
        
        if (result.Status == HealthStatus.Healthy)
        {
            return Ok(new { status = "ready" });
        }

        return StatusCode(503, new { status = "not ready" });
    }
}
